debs/SPECS/wazuh-agent/debian/postrm

```diff
@@ -51,6 +51,37 @@ case "$1" in
     ;;
 
     upgrade)
+        # If the upgrade downgrades to earlier versions, restore ownership
+        if command -v ${DIR}/bin/ossec-control > /dev/null 2>&1; then
+
+            OSMYSHELL="/sbin/nologin"
+
+            if [ -f ${DIR}/queue/sockets/.agent_info ]; then
+                mv ${DIR}/queue/sockets/.agent_info ${DIR}/queue/ossec/
+            fi
+
+            if ! getent group ossec > /dev/null 2>&1; then
+                addgroup --system ossec > /dev/null 2>&1
+            fi
+
+            if ! getent passwd ossec > /dev/null 2>&1; then
+                adduser --system --home /var/ossec --shell ${OSMYSHELL} --ingroup ossec ossec > /dev/null 2>&1
+            fi
+
+            find ${DIR} ! -group root -exec chgrp ossec {} \; > /dev/null 2>&1
```

Comment:
Some files have root group and/or user by default in a fresh install. Can you validate that this change allows Wazuh to execute all its functionalities?

Reply:
These lines take files that are _not_ owned by root (user nor group) and assigns them to ossec. These is only run in downgrades to older Wazuh versions.